Restructure installation guide into multiple sections

markpollack opened the pull request:
Fixes https://github.com/spring-cloud/spring-cloud-dataflow/issues/3205

cppwfs commented on content/documentation/variables.json:
```diff
@@ -3,5 +3,6 @@
     "thing1": "thing2"
-  }
+  },
+  "dataflow-version": "2.1.0.BUILD-SNAPSHOT"
```
Set this 2.1.0.M1 or all the `wget` commands will return a 404 because we don't have a BUILD-SNAPSHOT tag.

markpollack replied:
We need the equivalent of this in the old doc `<propertyregex property="github-tag" override="true" input="${github-tag}" regexp=".*SNAPSHOT" replace="master" />`

markpollack replied:
Damien will create a plugin, so will leave it as is.

cppwfs commented on content/documentation/pages/1-installation/1-local/1-docker.md:
````diff
@@ -0,0 +1,134 @@
+In the directory where you downloaded `docker-compose.yml`, start the system, as follows:
+
+```
+export DATAFLOW_VERSION={local-server-image-tag}
````
Can we use variables here so that the user can just do a copy paste?
Or remove them all together

markpollack replied:
Yes, this will require the same logic mentioned in the previous comment.  I will change to use `%local-server-image-tag%` and review the entire doc in another pass for the asciidoc style usage of variables.

cppwfs commented on content/documentation/pages/1-installation/1-local/1-docker.md:
```diff
@@ -0,0 +1,134 @@
+
+    $ docker exec -it dataflow-server java -jar shell.jar
+
+If you have started the Data Flow server via `java -jar`, download and start the shell using the following commands:
```
This line needs to be replaced since we are only discussing docker-compose deployment.   It could be replaced with something like, "You can run the Spring Cloud Data Flow shell without docker using the following instructions:"

markpollack replied:
Moved into the 3-manual section.

cppwfs commented on content/documentation/pages/1-installation/1-local/2-docker-customize.md:
```diff
@@ -0,0 +1,200 @@
+            volumes:
+              - /foo/bar/apps:/root/apps
+
+This configuration provides access to the `my-app.jar` (and the other files in the folder) from within container’s `/root/apps/` folder.
```
Instead of `my-app.jar`, would we rather say, "This configuration provides access to the directory(`/foo/bar`) containing your boot app jars, from within the container's `/root/apps` folder."

markpollack replied:
ok

cppwfs commented on content/documentation/pages/1-installation/1-local/2-docker-customize.md:
```diff
@@ -0,0 +1,200 @@
+    `spring-cloud-dataflow-grafana-prometheus` with
+    `spring-cloud-dataflow-grafana-influxdb`.
+
+## Accessing the host file system
```
For discussion.  In this section should we focus on mounting the .m2 since our examples lead us that direction.   Since to my knowledge we only show maven and docker resources and not file.

cppwfs commented on content/documentation/pages/1-installation/1-local/3-manual.md:
```diff
@@ -0,0 +1,104 @@
+1.  Download the Spring Cloud Data Flow Server by using the following
+    command:
+
+        wget https://repo.spring.io/{version-type-lowercase}/org/springframework/cloud/spring-cloud-dataflow-server/{project-version}/spring-cloud-dataflow-server-{project-version}.jar
```
Can we use variables here so that the user can just do a copy paste?

markpollack replied:
Changed reference of `{project-version}` to `%dataflow-version%`.   Added a placeholder variable for `version-type-lowercase`

cppwfs commented on content/documentation/pages/1-installation/1-local/3-manual.md:
```diff
@@ -0,0 +1,104 @@
+
+## Install messaging middleware
+
+These instructions require that RabbitMQ be running on the same machine as Skipper and the Spring Cloud Data Flow server and shell.
```
We defaulted to kafka in the docker compose.   Should we stick with Kafka?

markpollack replied:
There is a TODO to address this.

cppwfs commented on content/documentation/pages/1-installation/1-local/3-manual.md:
```diff
@@ -0,0 +1,104 @@
+    directory where you downloaded Skipper, run the server by using
+    `java -jar`, as follows:
+
+        $ java -jar spring-cloud-skipper-server-{skipper-version}.jar
```
We can remove the `$` symbol .  This should help with copy/paste

markpollack replied:
ok

cppwfs commented on content/documentation/pages/1-installation/1-local/3-manual.md:
````diff
@@ -0,0 +1,104 @@
+**TODO screen shot instead of shell command**
+
+```
+http://bit.ly/Einstein-SR2-stream-applications-kafka-maven
````
We can use a variable here as shown in this PR:
https://github.com/spring-io/dataflow.spring.io/pull/26/files#diff-becd5e9615be77785ac98fef42666f90R3-R4

markpollack replied:
This is just a placeholder for the TODO

cppwfs commented on content/documentation/pages/1-installation/1-local/3-manual.md:
```diff
@@ -0,0 +1,104 @@
+You can register applications using the UI or the shell.
+Even though we are only using two pre-built applications, we will register the full set of pre-built applications.
+
+Since local installation guide uses Kafka as the messaging middleware, register the Kafka version of the applications.
```
Above we tell the user to install rabbitmq but here we are discussing kafka.

markpollack replied:
this section prob needs to go somewhere else.  added bitly links for both now, will need to switch to dataflow.spring.io links once that is up and running.

cppwfs commented on content/documentation/pages/1-installation/2-cloudfoundry/0-index.md:
```diff
@@ -0,0 +1,10 @@
+
+# Installation
+
+This section show show to install Data Flow on Cloud Foundry.
```
show is duplicated

markpollack replied:
ok

cppwfs commented on content/documentation/pages/1-installation/2-cloudfoundry/1-cf-cli.md:
```diff
@@ -50,7 +50,7 @@ You can use `cf marketplace` to discover which plans are available to you, depen
 | into connection limits. This usually means you can't use any free plans.
 
-## Install on Cloud Foundry
+## Manifest based installation on Cloud Foundry
```
In this section, can we use variables here so that the user can just do a copy paste?

markpollack replied:
discussed it with sabby, felt it was making it too noisy and using manifests is more prevalent in practice. ATM users have to cut-n-paste into a manifest.yaml file.  Can link to a sample one in our github I guess.  Will leave it as is for now.

cppwfs commented on content/documentation/pages/1-installation/2-cloudfoundry/1-cf-cli.md:
```diff
@@ -1,14 +1,14 @@
-path: 'installation/cloudfoundry/'
-title: 'Cloud Foundry'
-description: 'Install Data Flow on your Cloud Foundry'
+path: 'installation/cloudfoundry/cf-cli'
```
For discussion.  Can we make this more concise, like Gunnar's example or is that the wrong direction.

markpollack replied:
What is gunnar's example?

cppwfs replied:
Located here: 
http://localhost:8000/documentation/master/batch-developer-guides/batch/data-flow-simple-task-cloudfoundry/

cppwfs commented on content/documentation/pages/1-installation/3-kubernetes/0-index.md:
```diff
@@ -0,0 +1,15 @@
+This guide also describes setting up an environment for testing Spring Cloud Data Flow on the Google Kubernetes Engine and is not meant to be a definitive guide for setting up a production environment.
+You can adjust the suggestions to fit your test setup.
+Remember that a production environment requires much more consideration for persistent storage of message queues, high availability, security, and other concerns.
+Refer to [this section](/documentation/installation/kubernetes/creating-a-cluster) for some high level guidance for creating a Kubernetes cluster.
```
Invalid URL

markpollack replied:
fixed

cppwfs commented on content/documentation/pages/1-installation/3-kubernetes/1-helm.md:
```diff
@@ -1,422 +1,10 @@
-
-    $ minikube service --url scdf-server
-    https://192.168.99.100:31991
+## Install using Helm
```
We can delete the "Helm Installation" title below.

cppwfs commented on content/documentation/pages/1-installation/3-kubernetes/1-helm.md:
```diff
@@ -1,422 +1,10 @@
-
-    $ minikube service --url scdf-server
-    https://192.168.99.100:31991
+## Install using Helm
```
Should we focus on only one installation option either helm or kubectl?   I'm thinking for simplicity.   A.   i.e. we use kubectl since its the easiest against minikube.   And refer them to the helm chart instructions if they need it.

markpollack replied:
If you are ok with installing helm, that is the easiest way, so I would prefer promoting that approach.  I don't want to get rid of the kubectl option either - in fact we may make it much easier by adding in skaffold and using the features in kustomize/kubectl.

cppwfs commented on content/documentation/pages/1-installation/3-kubernetes/2-kubectl.md:
```diff
@@ -0,0 +1,328 @@
+They have the required metadata set for service discovery needed by the different applications and services deployed.
+To check out the code, enter the following commands:
+
+    $ git clone https://github.com/{github-repo}
```
We can uses wget for the proper files instead of the git clone option.   David's example gives us an example of this.

markpollack replied:
there are many files, would need to have a .zip

cppwfs commented on content/documentation/pages/1-installation/3-kubernetes/2-kubectl.md:
```diff
@@ -0,0 +1,328 @@
+
+2. The Skipper service should be running and the `SPRING_CLOUD_SKIPPER_CLIENT_SERVER_URI` property in `src/kubernetes/server/server-deployment.yaml` should point to it.
+
+The Data Flow Server uses the [Fabric8 Java client
```
Probably don't need to go into this level of detail.   i.e. fabric8

markpollack replied:
I think we want to directly refer to the section - https://github.com/fabric8io/kubernetes-client#configuring-the-client so that users understand their options for configuration.

sabbyanandan commented:
Closed by mistake.

markpollack commented on content/documentation/pages/1-installation/1-local/2-docker-customize.md:
```diff
@@ -0,0 +1,200 @@
+          rabbitmq:
+            image: rabbitmq:3.7
+            expose:
+              - "5672"
```
This section is discussing what to replace in the docker compose file in order to use RabbitMQ by manually editing the file, so the user will be deleting everything under
```
kafka:
  image: confluentinc/cp-kafka:5.2.1
```

I'll make the description of what to delete shorter.

markpollack commented on content/documentation/pages/1-installation/1-local/2-docker-customize.md:
```diff
@@ -0,0 +1,200 @@
+## Accessing the host file system
+
+If you are developing custom applications on your local machine, you will need to register them with Data Flow.
+Data Flow is running inside of a Docker container, so the Docker container will will need access to your local file system in order to resolve the registration reference.
```
ok

markpollack commented on content/documentation/pages/1-installation/1-local/2-docker-customize.md:
```diff
@@ -0,0 +1,200 @@
+
+    dataflow:>app register --type source --name my-app --uri file://root/apps/my-app-1.0.0.RELEASE.jar
+
+[[tip]] | Metadata URIs]]
```
ok

markpollack commented:
addressed PR comments and merged into master myself since there were some tedious conflicts to resolve.

pushed as bcdc26551ea4966d3d7c1a4f2f76e3ca680e28e8
